src/vos/vos_dtx.c

```diff
@@ -311,11 +311,14 @@ dtx_act_ent_update(struct btr_instance *tins, struct btr_record *rec,
 	struct vos_dtx_act_ent	*dae_old;
 
 	dae_old = umem_off2ptr(&tins->ti_umm, rec->rec_off);
-	if (DAE_EPOCH(dae_old) != DAE_EPOCH(dae_new))
-		D_ASSERTF(0, "NOT allow to update act DTX entry for "DF_DTI
+	if (DAE_EPOCH(dae_old) != DAE_EPOCH(dae_new)) {
+		D_ASSERTF(!dae_old->dae_prepared,
+			  "NOT allow to update act DTX entry for "DF_DTI
```

Comment:
"updating active DTX entry for DF_DTI is not allowed"

or 

"Disallow update to active..."